File: 258-add-digits/add-digits.java
class Solution {
    public int addDigits(int num) {
        String string=String.valueOf(num);
        if (string.length()==1){
            return num;
        }       
        int summ=0;
        for(char c:string.toCharArray()){
            summ=summ+Character.getNumericValue(c);
        }
        return addDigits(summ);
        }
}

File: 258-add-digits/add-digits.py
class Solution:
    def addDigits(self, num: int) -> int:
        string=str(num)
        if len(string)==1:
            return num
        summ=0
        for i in string:
            summ=summ+int(i)
        return self.addDigits(summ)



        